tests/ui/target_modifiers/defaults_check.rs

```diff
@@ -0,0 +1,20 @@
+// Tests that default unspecified target modifier value in dependency crate is ok linked
+// with the same value, explicitly specified
+//@ aux-crate:default_reg_struct_return=default_reg_struct_return.rs
+//@ compile-flags: --target i686-unknown-linux-gnu -Cpanic=abort
```

Comment:
Question: all of these tests specifically focus on `i686-unknown-linux-gnu`. Are there meaningful differences for e.g. windows-msvc or windows-gnu or aarch64-apple that may warrant additional test coverage of this mechanism?

Reply:
No, it is just because example flags `reg-struct-return` and `regparm` exist for x86.

Reply:
`I think having a specific cli flag interface check would be good (to check what the flag accepts vs rejects).`
Didn't get it. Could you be more specific, please? Some examples, maybe.

Reply:
Sure. What we mean is to have a specific revisioned cli interface ui test that tries to exercise the `-Cunsafe-allow-abi-mismatch` flag, to check:

- That a valid value is accepted: e.g. `-Cunsafe-allow-abi-mismatch=reg-struct-return`
- What happens if the user does not specify a value? e.g. `-Cunsafe-allow-abi-mismatch` alone.
- What happens if the user specifies *multiple* values? e.g. `-Cunsafe-allow-abi-mismatch=reg-struct-return,regparm`.
- What happens if the user specifies an *invalid* value? e.g. `-Cunsafe-allow-abi-mismatch=toaster` (this is not a thing).

Reply:
Yes, one flag and absent value was already tested before. I have added test case with two flags and invalid value. Invalid flag is now processed as an error: 
```
error: unknown target modifier `unknown_flag`, requested by `-Cunsafe-allow-abi-mismatch=unknown_flag`
```